ci-operator/step-registry/osd-ccs/fleet-manager/sc/provision/osd-ccs-fleet-manager-sc-provision-commands.sh

```diff
@@ -4,28 +4,95 @@ set -o nounset
-#else
-#  echo "Did not find compatible cloud provider cluster_profile"
-#  exit 1
-#fi
-
-
-# Log in
-OCM_VERSION=$(ocm version)
-OCM_TOKEN=$(cat "${CLUSTER_PROFILE_DIR}/ocm-token")
-echo "Logging into ${OCM_LOGIN_ENV} with offline token using ocm cli ${OCM_VERSION}"
-ocm login --url "${OCM_LOGIN_ENV}" --token "${OCM_TOKEN}"
+AWSCRED="${CLUSTER_PROFILE_DIR}/.awscred"
+if [[ -f "${AWSCRED}" ]]; then
+  export AWS_SHARED_CREDENTIALS_FILE="${AWSCRED}"
+  export AWS_DEFAULT_REGION="${OSDFM_REGION}"
+else
+  echo "Did not find compatible cloud provider cluster_profile"
+  exit 1
+fi
 
-# Log in
+# Log in with OSDFM token
 OCM_VERSION=$(ocm version)
-OCM_TOKEN=$(cat "${CLUSTER_PROFILE_DIR}/fleetmanager-token")
+OSDFM_TOKEN=$(cat "${CLUSTER_PROFILE_DIR}/fleetmanager-token")
 echo "Logging into ${OCM_LOGIN_ENV} with offline token using ocm cli ${OCM_VERSION}"
-ocm login --url "${OCM_LOGIN_ENV}" --token "${OCM_TOKEN}"
+if [[ ! -z "${OSDFM_TOKEN}" ]]; then
+  echo "Logging into ${OCM_LOGIN_ENV} with osdfm offline token"
+  ocm login --url "${OCM_LOGIN_ENV}" --token "${OSDFM_TOKEN}"
+  if [ $? -ne 0 ]; then
+    echo "Login failed"
+    exit 1
+  fi
+else
+  echo "Cannot login! You need to specify the offline token OSDFM_TOKEN!"
+  exit 1
+fi
+
+# Check if SC already exists
+cluster_num=$(ocm get /api/osd_fleet_mgmt/v1/service_clusters -p search="region='${OSDFM_REGION}'" | jq -r '.size')
+if ((${cluster_num} > 0)); then
+  CLUSTER_ID=$(ocm get /api/osd_fleet_mgmt/v1/service_clusters -p search="region='${OSDFM_REGION}'" | jq -r '.items[0].id')
+  echo "Service Cluster '${CLUSTER_ID}' already exists in region '${OSDFM_REGION}', exit"
+  exit 1
+fi
+
+# Create SC
+echo '{"region": "'${OSDFM_REGION}'", "cloud_provider": "aws"}' | ocm post /api/osd_fleet_mgmt/v1/service_clusters
+
+# Check if Service Cluster is ready
+echo "Waiting for Service Cluster ready..."
+wait_for_cluster "service_clusters"
+
+# Save Service Cluster info
+CLUSTER_ID=$(ocm get /api/osd_fleet_mgmt/v1/service_clusters -p search="region='${OSDFM_REGION}'" | jq -r '.items[0].id')
+echo "${CLUSTER_ID}" >> "${SHARED_DIR}/osd-fm-sc-id"
+echo "Service Cluster:${CLUSTER_ID}"
+
+# Save Management Cluster info, there might be multiple MC
+cluster_num=$(ocm get /api/osd_fleet_mgmt/v1/management_clusters -p search="region='${OSDFM_REGION}'" | jq -r '.size')
+if ((${cluster_num} >= 1)); then
+    for((i=0;i<${cluster_num};i++));
+    do
+        cluster_state=$(ocm get /api/osd_fleet_mgmt/v1/management_clusters -p search="region='${OSDFM_REGION}'" | jq -r '.items['$i'].status')
+        echo "Index $i MC is in ${cluster_state} state"
+        if [[ "${cluster_state}" == "ready" ]]; then
+            mc_ready_idx=$i
+        fi
+        CLUSTER_ID=$(ocm get /api/osd_fleet_mgmt/v1/management_clusters -p search="region='${OSDFM_REGION}'" | jq -r '.items['$i'].id')
+        echo "${CLUSTER_ID}" >> "${SHARED_DIR}/osd-fm-mc-id"
+        echo "Management Cluster:${CLUSTER_ID}"
+    done
+fi
```

Comment:
How do you treat the event if there is no MC ready?

Reply:
Done.Use the response of OCM post command and save it immediately.